assets/src/apps/delivery/components/TextParser.ts

```diff
@@ -8,7 +13,7 @@ export const templatizeText = (
   env?: Environment,
   isFromTrapStates = false,
 ): string => {
-  let innerEnv = env;
+  let innerEnv = env || defaultGlobalEnv;
```

Comment:
I don't think we can import the defaultGlobalEnv into here for use in externalActivity, I think this is why I'm sending env reference in init or context change (or both). when you import them, they become part of a bundle, and the components are their *own* web component bundle, it won't be the same reference.

Reply:
if you need to have a default here, which I don't think you do, then you might as well just say `|| new Environment()`

Reply:
Oops, I didn't knew that env reference was passed in part components. Changes made and PR Updated. followed the same patterns as of text flow part component.